File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    result = []
    score = [0, 0, 0]
    person1 = [1, 2, 3, 4, 5]
    person2 = [2, 1, 2, 3, 2, 4, 2, 5]
    person3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    for idx, answer in enumerate(answers):
        if answer == person1[idx % len(person1)]:
            score[0] += 1
        if answer == person2[idx % len(person2)]:
            score[1] += 1
        if answer == person3[idx % len(person3)]:
            score[2] += 1
    
    for idx, s in enumerate(score):
        if s == max(score):
            result.append(idx+1)

    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
  var answer = [];
  const p1 = [1, 2, 3, 4, 5];
  const p2 = [2, 1, 2, 3, 2, 4, 2, 5];
  const p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

  let score1 = answers.filter((ans, idx) => ans === p1[idx % p1.length]).length;
  let score2 = answers.filter((ans, idx) => ans === p2[idx % p2.length]).length;
  let score3 = answers.filter((ans, idx) => ans === p3[idx % p3.length]).length;
  const max = Math.max(score1, score2, score3);

  if(score1 === max) answer.push(1);
  if(score2 === max) answer.push(2);
  if(score3 === max) answer.push(3);
    
  return answer;
}